### PR TITLE
[Windows] Fixes border content clipping

### LINKED ISSUE #16444: Border's corner radius applied to child's top left corner in Windows, iOS and MacCatalyst
### Description

Parent border's corner radius is applied to first child's(border/rounded rectangle) top left corner

### Steps to Reproduce

Just copy the following code and execute
`<Border StrokeShape="RoundRectangle 12"
                    Stroke="blue"
                    BackgroundColor="LightBlue"
                    Padding="16">
                <Grid RowDefinitions="Auto,Auto">
                    <Border HeightRequest="200"
                            BackgroundColor="DarkGreen">
                        <Label Text="Sample"
                               FontSize="16" />
                    </Border>
                    <Label Text="Sample"
                           Grid.Row="1"
                           FontSize="21" />
                </Grid>
            </Border>`
Border_Corner_Radius

### Link to public reproduction project repository

https://github.com/satzmuthu/TestMAUI

### Version with bug

7.0.49

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Windows, macOS

### Affected platform versions

iOS 16, Window 10.0.19041.0

### Did you find any workaround?

Used frame  instead of parent border

### Relevant log output

_No response_

### LINKED ISSUE #17109: [Windows] Border content is not correctly clipping and aligning
### Description

On Windows, content in borders do not correctly adapt to the shape of the border and also does not correctly align:
Clipping issue

### Steps to Reproduce

1. Run the Maui.Sample.Controls app
2. Navigate to Controls -> Border -> Border resize Content
3. Move the slider and check how the content adapts to each border

Expected: the content should not go outside the border bounds, and the + sign should always be centered.

### Link to public reproduction project repository

_No response_

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### LINKED ISSUE #17500: [Windows] Border with rounded corners does not work well
### Description

The border component when used with rounded corners does not work correctly on Windows. First it cuts the content from the left/top margin, then it lets the content exceed the limits of the right margin.
BorderError fw

### Steps to Reproduce

1. Create MAUI Project any version (versions 6-8.0rc).
2. In MainPage add border with StrokeShape with Round Rectangle, example:
```
        <Border StrokeShape="RoundRectangle 50" HeightRequest="100" WidthRequest="200" BackgroundColor="White">
            <Label Text="Hello World! I love you!" />
        </Border>
        <Border StrokeShape="RoundRectangle 50" Padding="40, 0,0,0" HeightRequest="100" WidthRequest="200" BackgroundColor="White">
            <Label Text="Hello World! I love you! Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!Hello World! I love you!" />
        </Border>
        <Border StrokeShape="RoundRectangle 50" Padding="40" HeightRequest="100" WidthRequest="200" BackgroundColor="White">
            <Label Text="Hello World! I love you!" />
        </Border>
```
3. Run App in Windows

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.1.9171

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

WinUI 3

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by emaf**
### Description of Change

Fixes clip path size to consider the border stroke thickness, so it only uses the inner space. For the space the border takes we consider both sides (top and bottom, or left and right) and not just the thickness.

On top of that, the clip should not be sentered with the content, so we need to offset it based on the contents position.

### Issues Fixed

Fixes #17109
Fixes #16444
Fixes #17500

### CODE REVIEW

**Comment by emaf** [comment]
This PR partially fixes how content in borders with shapes other than rectangles clip, but there are still two problems:
1. When the border shape is a polygon, the content goes over the border. The problem seems to be the Polygon's shape GetPath method is not considering the bounds set on GetPathForBounds.
2. When scaling images, the clip seems to be scaled too.

clipping problems

**Comment by emaf** [comment]
> 1. When the border shape is a polygon, the content goes over the border. The problem seems to be the Polygon's shape GetPath method is not considering the bounds set on GetPathForBounds.

Polygon just returns the original path: https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Shapes/Polygon.cs#L53-L68. And this is something that happens with other shapes too, like polyline https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Shapes/Polyline.cs.

On the other hand, ellipse does consider the bounds through Width/HieghtForPathComputation https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Shapes/Ellipse.cs#L30-L45, which maks it work correctly.

**Comment by samhouts** [comment]
Is this ready for review?

**Comment by jstedfast** [comment]
Filed the Image clipping issue here: https://github.com/dotnet/maui/issues/17523

**Review by jstedfast — DISMISSED** [review]

File: src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.csproj [hunk 1]
```diff
@@ -45,6 +45,18 @@
     <MauiSplashScreen Include="Resources\Splash\splash.svg" Color="#FFFFFF" BaseSize="168,208" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Update="Elements\Border1.xaml.cs">
```
Comment (PureWeen) [inline src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.csproj:49]: I think we can remove these

Comment (BretJohnson) [reply src/Controls/samples/Controls.Sample.UITests/Controls.Sample.UITests.csproj]: Removed (thanks to jeff earlier)

**Comment by BretJohnson** [comment]
Fully enabling the tests is now moved a separate PR, here: https://github.com/dotnet/maui/pull/17633, so that we can merge this PR ASAP.

**Review by BretJohnson — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**